### ISSUE TITLE
About the Depth

### ISSUE THREAD

**Issue by jypjypjypjyp** [cvg/depthsplat#30]
Hello, I've noticed that the calculation of the center of the Gaussian (mean) in the code is achieved by multiplying the direction by the depth. 
```
origins, directions = get_world_rays(coordinates, extrinsics, intrinsics)
means = origins + directions * depths[..., None]
```
However, this is different from what I consider as depth. As shown in the figure below, I think the depth is the distance from the image plane to the camera (**Z**).
![image](https://github.com/user-attachments/assets/f0ecda5a-defd-44f6-a147-88d659ff8f18)

When I was running some experiments, I found that for the flat objects on the depth map, the point cloud according to the Gaussian center points showed an obvious curvature.
![image](https://github.com/user-attachments/assets/60ed5a14-bdc1-4308-8cf6-ee8c32e80d92)
![image](https://github.com/user-attachments/assets/bc6fb2fa-52de-4ca5-b9ef-b9a060c62dda)
![image](https://github.com/user-attachments/assets/6563d0f7-8de4-4706-a064-469f643f6c6b)

May I ask if this result is correct?

**Comment by jypjypjypjyp** [comment]
> Hi, thank you for your experiments. Yes, I think your comments make sense and this implementation is inherited from pixelsplat's codebase and please refer to [here](https://github.com/dcharatan/pixelsplat/issues/82) for some related discussions. You can try to replace the code with
> 
> ```
> directions = directions / directions[..., -1:]
> ```
> 
> to see how the results differ.

@haofeixu 
The two series of models provided - the depth model and the novel view synthesis model - are quite confusing in that the depth model uses Z for depth, while the depth in the novel view synthesis model is represented by distance.
Moreover, it is stated in the paper that the depth task and the novel view synthesis task can improve each other. Then why not provide a unified model that can handle both the depth task and the novel view synthesis task? Separating these two tasks goes against the argument in the paper. Additionally, in my experiments, I found that the depth accuracy of the novel view synthesis model in low-texture areas is far inferior to that of the depth model.

**Comment by ZhongmiaoYan** [comment]
Hi @haofeixu and @jypjypjypjyp,

I encountered a similar issue in the depth prediction model and was wondering if this issue has been clarified or resolved on your side? It seems that in the code, depth is referred to as the z-coordinate, but when generating the Gaussian point cloud center based on the predicted depth values, the model seems to interpret depth as the distance from the point to the camera center.

Typically, learning-based methods adapt to the specific depth settings, but I am uncertain whether the coexistence of these two different settings might have any impact on the model’s performance.

@jypjypjypjyp, I’m also very intrigued by your mention of working on two tasks together. Do you have any thoughts on how these tasks could be more closely integrated?

Thank you for your time and insights!

**Comment by haofeixu** [comment]
Hi @jypjypjypjyp @ZhongmiaoYan, thank you for your questions. 

Regarding the depth unprojection for 3DGS, I agree with you that using z-depth makes more sense. Our unprojection implementation is inherited from pixelsplat's codebase and thus is a legacy issue, please kindly refer to [here](https://github.com/dcharatan/pixelsplat/issues/82) for related discussions. I can confirm that using z-depth for unprojection in our model leads to further improved results and we plan to fix this legacy issue in the near future.

Regarding separate depth and view synthesis models, we would like to note that the primary focus of our paper is to study the cross-task transfer properties with different pre-trained model weights (similar to the setup of [MAE](https://arxiv.org/abs/2111.06377) which first performs pre-training and then studies the transferring performance on downstream tasks). Training a single model for both tasks is more similar to the multi-task learning setup, where there are [additional challenges](https://arxiv.org/abs/1905.07553) to consider (e.g., how to balance the training of different tasks, different training samples from different datasets etc.). So far we observe that training separate models leads to the best performance for both tasks while we can leverage the pre-trained weights from another task as initialization (i.e., cross-task transfer). 

Please also kindly note that the 3DGS representation is mainly designed for rendering and the learned 3D Gaussians might not be precisely aligned with the actual surface, which potentially leads to different behaviors of the depth models (initial pre-trained depth vs. final depth fine-tuned with the rendering loss). This phenomenon can also be observed in per-scene optimization methods, where the optimized Gaussian centers might have different distributions from the point cloud from SfM at initialization.

To summarize, separate depth and view synthesis models can achieve the best performance (so far) for both tasks, and training a joint model is possible but would need to consider the additional challenges mentioned above.

Thank you again for your questions and I am open to further discussions.

**Comment by ZhongmiaoYan** [comment]
Thank you for your prompt response. As you mentioned in other issues, relying solely on depth prediction from a single frame may lead to depth values that don't match the true physical scale, as well as a lack of consistency across multiple frames. This can affect various 3D reconstruction methods, including 3DGS.

However, I noticed that you used the poses of the images in your work. Perhaps we could try a multi-frame constrained depth estimation model to improve the consistency of depth estimates and recover the true physical scale. This could help explore whether such a depth prior could enhance the Gaussian reconstruction and rendering results. Do you think my idea is feasible and worth trying?
